wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/mecanumcontrollercommand/RobotContainer.java

```diff
@@ -121,10 +121,10 @@ public Command getAutonomousCommand() {
             m_robotDrive::setDriveMotorControllersVolts, // Consumer for the output motor voltages
             m_robotDrive);
 
-    // Reset odometry to the starting pose of the trajectory.
-    m_robotDrive.resetOdometry(exampleTrajectory.getInitialPose());
-
-    // Run path following command, then stop at the end.
-    return mecanumControllerCommand.andThen(() -> m_robotDrive.drive(0, 0, 0, false));
+    // Reset odometry to the initial pose of the trajectory, Run path following command, then stop
+    // at the end.
+    return mecanumControllerCommand
+        .andThen(() -> m_robotDrive.drive(0, 0, 0, false))
+        .beforeStarting(() -> m_robotDrive.resetOdometry(exampleTrajectory.getInitialPose()));
```

Comment:
This ordering is correct, but I think it would be clearer as (pseudocode):
```
runOnce(resetOdometry)
  .andThen(mecanumControllerCommand)
  .andThen(stop)
```